packages/xstate-react/src/useActor.ts

```diff
@@ -56,9 +55,7 @@ export function useActor<TLogic extends AnyActorLogic>(
     actorRef.start();
 
     return () => {
-      actorRef.stop();
-      actorRef.status = InterpreterStatus.NotStarted;
-      (actorRef as any)._initState();
+      actorRef.reset();
```

Comment:
to improve the story for offscreen components this should likely be something like:
```suggestion
      actorRef.reset(actorRef.getPersistedState());
```